packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -104,6 +129,38 @@ export function useGridEditRows(
     [apiRef, forceUpdate, logger, setGridState],
   );
 
+  const setRowMode = React.useCallback(
```

Comment:
To type the arguments and the return type, I propose to always do the following :
I avoids inconsistencies between the public API signatures and the method signature.
For instance in another hook the public API was setting an argument as mandatory and the method was setting it as optional. TypeScript was not complaining since the public API was stricter.

x other methods

```suggestion
  const setRowMode = React.useCallback<GridEditRowApi['setRowMode']>(
```